### ISSUE TITLE
how to make nested facets

### ISSUE THREAD

**Issue by NoraGithub** [antvis/g2#255]
Thanks for your great job of G2.
I wonder whether there's a way to make nested fecets. As far as I read the docs, the only four types like `rect` ,`list` and so on cannot be nested.

Take a dataset as an example:
```
[{'chart_type': 'bar',
  'index': 'sum_人头',
  'sum_人头': 70.0,
  '地区': '广东',
  '性别': '女',
  '兴趣':'美食',
  '分类':'小资',
},
...
```
In the example , I have, `地区`,`性别`,`兴趣`,`分类`,4 dimensions in total, and I can only make a `rect` by 2 dimensions or `list` by 1 dimension. 
What I need is like that first make `rect `facet by [`地区`,`性别`] and inside the view make `list` facet by [`兴趣`] again.

Hope I am asking a clear question.

**Comment by NoraGithub** [comment]
My way is now turn a `view` into a `chart` to `facet`again by 
```
...
eachView(view,facet){
          view.__proto__ = G2.Chart.prototype;
          view.facet('list',
....
```
However, I don't think it's right because of bugs in console. Would you please give some instructions?

**Comment by leungwensen** [comment]
Sorry, but this is not supported now. It is not actually a common case to manage facets in a `hierarchy` structure, so it seems not suitable to implement this feature upon the basement of `the grammar of graphics`.

Though facet is a matter of data splitting and space distributing, it can be done manually.

**Comment by lelandwilkinson** [comment]
Not sure if this is useful, but here's some Java code I developed to parse
GG algebra to canonical form. This parser simplifies the processing of more
complex GG expressions. I imagine it wouldn't be too hard to translate it
into JavaScript.

Leland Wilkinson

**Comment by leungwensen** [comment]
@lelandwilkinson Thanks a lot! Could we get a copy of the code?

antv@antfin.com

**Comment by dxq613** [comment]
We originally designed the nesting of views in the original design, but in the business practice we found that the user's window is too small, with the addition of axes and so on, so we removed the view's nesting and only allowed the chart to contain multiple views.

**Comment by NoraGithub** [comment]
@dxq613  In my option, in some scene, it's necessary, though it's to tell the user that  "hey, I cannot  hold that much data".(We are designing a flexible tool)

The importance of nested char is that, it made a complete Grammar of Graphics(in my knowledge), and it's also the biggest difference between g2 and echart.

**Comment by weepy3641** [comment]
@dxq613 
Nested axis、scroll bar is often be used to resolve the problem of space in nested facets chart.
like tableau.

Left axis is nested facet axis.
![image](https://user-images.githubusercontent.com/3838676/33927655-11c32df2-e01e-11e7-868f-baf612e8d9e9.png)

Bottom scroll bar can move chart geometry area left to right.
![image](https://user-images.githubusercontent.com/3838676/33927849-b6761724-e01e-11e7-9cfb-7ae5c758aef9.png)

**Comment by leungwensen** [comment]
Thanks all! Nested facet will be included in a future version.

**Comment by troy351** [comment]
any update on this one?

**Comment by leungwensen** [comment]
@troy351 this feature is now WIP, it will be released in G2 4.0. Now it is being tested in an internal BI system.

**Comment by manV** [comment]
Thanks for implementing this. Is there any example or docs or piece of code I can look at that uses such nesting of facets?
